doc/reference/reference_sql/sql_statements_and_clauses.rst

```diff
@@ -620,140 +620,95 @@ causes no error message, although (s1, s1) is probably a user error.
+    Then, insert data to this table:
 
-   ROLLBACK -- do transaction rollback and return an error.
+    ..  literalinclude:: /code_snippets/test/sql/foreign_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_parent_start
+        :end-before: insert_parent_end
+        :dedent:
 
-   The order of constraint evaluation is described in section Order of Execution in Data-Change Statements.
+2.  Create a child ``book`` table whose ``author_id`` column references the ``id`` column from the ``author`` table:
 
-   For example, suppose a new table  t has one column and the column has a unique constraint.
-   A transaction starts with START TRANSACTION.
-   The first statement in the transaction is INSERT INTO t VALUES (1), (2);
-   that is, "insert 1, then insert 2" -- Tarantool processes the new rows in order.
-   This statement always succeeds, there are no constraint violations.
-   The second SQL statement is INSERT INTO t VALUES (3), (2), (5);
-   that is, "insert 3, then insert 2".
-   Inserting 3 is not a problem, but inserting 2 is a problem -- it would violate the UNIQUE constraint.
+    ..  literalinclude:: /code_snippets/test/sql/foreign_key_table_constraint_test.lua
+        :language: sql
+        :start-after: create_child_start
+        :end-before: create_child_end
+        :dedent:
 
-   If behavior is ABORT: the second statement is rolled back, there is an error message. The table now contains (1), (2).
+    Insert data to the ``book`` table:
 
-   If behavior is FAIL: the second statement is not rolled back, there is an error message. The table now contains (1), (2), (3).
+    ..  literalinclude:: /code_snippets/test/sql/foreign_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_child_start
+        :end-before: insert_child_end
+        :dedent:
 
-   If behavior is IGNORE: the second statement is not rolled back, the (2) is not inserted, there is no error message. The table now contains (1), (2), (3), (5).
+3.  Check how the created foreign key constraint enforces data integrity.
+    The following error is raised on an attempt to insert a new book with the ``author_id`` value that doesn't exist in the parent ``author`` table:
 
-   If behavior is REPLACE: the second statement is not rolled back, the first (2) is replaced by the second (2), there is no error message. The table now contains (1), (2), (3), (5).
+    ..  literalinclude:: /code_snippets/test/sql/foreign_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_error_start
+        :end-before: insert_error_end
+        :dedent:
 
-   If behavior is ROLLBACK: the statement is rolled back, and the first statement is rolled back,
-   and there is an error message. The table now contains nothing.
+    On an attempt to delete an author that already has books in the ``book`` table, the following error is raised:
```

Comment:
The error shown is actually not what I expected from the description. This is a new limitation for FKs: referencing fields must now have an index in order to remove anything from the referenced space. Otherwise, we will not be able to remove from the referenced space, even if the row is not referenced. For example, if we add `(3, 'Alexander Pushkin')` to the `author` space and do not add anything to the `book` space, we won't be able to remove the added row. To remove it, we need to create an index on `book.author_id`.

To receive expected error I suggest to replace
```
        box.execute([[
            -- create_child_start
            CREATE TABLE book (
                id INTEGER PRIMARY KEY,
                title VARCHAR(200) NOT NULL,
                author_id INTEGER NOT NULL,
                FOREIGN KEY (author_id)
                    REFERENCES author (id)
            );
            -- create_child_end
        ]])
```
by
```
        box.execute([[
            -- create_child_start
            CREATE TABLE book (
                id INTEGER PRIMARY KEY,
                title VARCHAR(200) NOT NULL,
                author_id INTEGER NOT NULL UNIQUE,
                FOREIGN KEY (author_id)
                    REFERENCES author (id)
            );
            -- create_child_end
        ]])
```
and mention why we add UNIQUE constraint here.

Then the error will be:
```'Foreign key ''fk_unnamed_BOOK_1'' integrity check failed: tuple is referenced'```